openscap_daemon/system.py

```diff
@@ -640,6 +640,17 @@ def generate_report_for_task_result(self, task_id, result_id):
             self.config
         )
 
+    def generate_fix_for_task_result(self, task_id, result_id, fix_type):
+        task = None
+        with self.tasks_lock:
+            task = self.tasks[task_id]
+
+        return task.generate_fix_for_result(
```

Comment:
Yeah, you are right. My memory on how openscap-daemon works is fuzzy by now :-D tasks_lock locks the list of tasks, once you get a task out you don't have to keep the lock.